CMakeLists.txt

```diff
@@ -10,6 +10,14 @@
 cmake_minimum_required(VERSION 3.3.2 FATAL_ERROR)
 project(UJIT C)
 
+if(CMAKE_SYSTEM_NAME STREQUAL "Linux")
+  set(UJIT_OS "UJIT_OS_LINUX")
+elseif(CMAKE_SYSTEM_NAME STREQUAL "Darwin")
+  set(UJIT_OS "UJIT_OS_OSX")
```

Comment:
Not 100% sure on "UJIT_OS_OSX", maybe let's use "UJIT_OS_MACOS", if it's macOS now, not Mac OS X?

Reply:
According to Wikipedia, it is indeed macOS nowadays. Since we do not support any "true" Mac OS X, renamed everywhere where felt appropriate (e.g. the value of `jit.os` is intentionally left aligned with LuaJIT).